### PR TITLE
Fix for "undefined method `call` for class `Symbol`" exception

### PR DESCRIPTION
**Pull request by yogeshjain999**
**Issue**

We're getting an exception when `output_with_outer_ctx: true` is passed to a step which has symbol output.

```ruby
step :auth, output: :auth_output, output_with_outer_ctx: true

# Exception being thrown

# NameError: undefined method `call' for class `Symbol'
# Did you mean?  caller
#    lib/trailblazer/activity/dsl/linear/feature/variable_mapping/dsl.rb:192:in `method'
```

**Fix**

Calculate for `arity` based on `:output`'s type (Symbol, Proc, method etc)

### CODE REVIEW

File: lib/trailblazer/activity/dsl/linear/feature/variable_mapping/dsl.rb [hunk 1]
```diff
@@ -188,17 +188,21 @@ class Out < Tuple
               class FiltersBuilder
                 def self.call(user_filter, with_outer_ctx:, **options)
                   if with_outer_ctx
-                    callable    = user_filter # FIXME: :instance_method
-                    call_method = callable.respond_to?(:arity) ? callable : callable.method(:call)
+                    callable = Trailblazer::Option(user_filter) # FIXME: :instance_method
+                    arity = case user_filter
+                            when Symbol then nil
```
Comment (apotonick) [inline lib/trailblazer/activity/dsl/linear/feature/variable_mapping/dsl.rb:193]: We could extract that to something `Deprecate.option_arity`, I've needed this before (but will do when needed).

File: test/docs/variable_mapping_test.rb [hunk 2]
```diff
@@ -151,10 +151,10 @@ module E
         module Memo; end
         #:no-input
         class Memo::Create < Trailblazer::Activity::Path
-          step :authorize, output: :authorize_output
+          step :authorize, output: :authorize_output, output_with_outer_ctx: true
```
Comment (apotonick) [inline test/docs/variable_mapping_test.rb:154]: Could you add that as a separate test, please? From the diff it looks as if we're changing an existing test here to use an additional feature.

**Review by apotonick — CHANGES_REQUESTED** [review]

File: lib/trailblazer/activity/dsl/linear/feature/variable_mapping/dsl.rb (hunk 1 above)
Comment (yogeshjain999) [reply lib/trailblazer/activity/dsl/linear/feature/variable_mapping/dsl.rb:193]: Agree, should we do it now or separately ?

Comment (apotonick) [reply lib/trailblazer/activity/dsl/linear/feature/variable_mapping/dsl.rb:193]: Later... :laughing:

**Review by apotonick — APPROVED** [review]

**Pull request merged**